synapse/util/threepids.py

```diff
@@ -48,3 +50,23 @@ def check_3pid_allowed(hs, medium, address):
         return True
 
     return False
+
+
+def canonicalise_email(address) -> str:
+    """'Canonicalise' email address
+    Case folding of local part of email address and lowercase domain part
+    See MSC2265, https://github.com/matrix-org/matrix-doc/pull/2265
+
+    Args:
+        address (str): email address within that medium (e.g. "wotan@matrix.org")
+    Returns:
+        (str) The canonical form of the email address
+    Raises:
+        SynapseError if the number could not be parsed.
+    """
+
+    try:
+        address = address.split("@")
+        return address[0].casefold() + "@" + address[1].lower()
```

Comment:
what if the email address has two `@`s in it? Please study [CVE-2019-11340](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-11340) :)

Reply:
I have made an update.